### PR TITLE
Changed article page from using css grid to instead using column count

### LINKED ISSUE #6327: [Publications] Summary Link Order
There's been feedback for adjusting the summary link order or format such as:

- [Data for Empowerment](https://foundation.mozilla.org/en/data-futures-lab/data-for-empowerment/) research series, they want to be able to change the two columns to be vertically ordered not horizontal
- [Trustworthy AI whitepaper](https://foundation.mozilla.org/en/insights/trustworthy-ai-whitepaper/), Sarah W wanted to be able to have one column for the summary content https://github.com/mozilla/foundation.mozilla.org/issues/5800#issuecomment-744577728

Are these possible options to provide?

--- 

Updates: according to this [comment](https://github.com/mozilla/foundation.mozilla.org/issues/6327#issuecomment-801531125)

> have one column when there are less than 5 links, and two columns when there is more

### PR DESCRIPTION
**Pull request by danielfmiranda**
Closes #6327

**Link to sample test page with extra long link**: [link](https://foundation-s-6327-summa-fj8xlv.herokuapp.com/en/publication-page-with-child-article-pages/these-than-most-agree/)
**Link to sample test page with > 5 links**: [link](https://foundation-s-6327-summa-fj8xlv.herokuapp.com/en/publication-page-with-child-article-pages/health-threat-baby-world-gun/)
**Link to sample test page with < 5 links**:  [link](https://foundation-s-6327-summa-fj8xlv.herokuapp.com/en/publication-page-with-child-article-pages/officer-less-human-member-green-test/)

## Checklist

**Changes in Models:**
- [x] [Are my changes backward-compatible]

### CODE REVIEW

File: network-api/networkapi/wagtailpages/templates/wagtailpages/article_page.html [hunk 1]
```diff
@@ -28,7 +28,7 @@
             <div class="article-summary-title body-small">{% trans "SUMMARY" %}</div>
             <div class="article-links {% if get_titles|length > 5 %}article-links__two-col{% endif %}">
               {% for slug, header in get_titles %}
-                <a href="#{{ slug }}" class="article-summary-link">{{ header }}</a>
+                <a href="#{{ slug }}" class="article-summary-link {% if get_titles|length > 5 %}article-summary-link__two-col{% endif %}">{{ header }}</a>
```
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/article_page.html:31]: Checking if the page has more than 5 links, if so, assign "two-col" class.

File: source/sass/components/article-page.scss [hunk 2]
```diff
@@ -56,16 +52,20 @@
...
 
   margin-top: 17px;
   flex-basis: calc(50% - var(--spacing));
   flex-grow: 0;
   flex-shrink: 0;
-
   @include hover-focus-active {
     color: inherit;
   }
+
```
Comment (danielfmiranda) [inline source/sass/components/article-page.scss:65]: if there are less than 5 links, the article summary links display block one after the other.

However if there are more than 5 and the two-col class is assigned, they display inline-block to stay aligned with one another in the two column format.

File: source/sass/components/article-page.scss [hunk 3]
```diff
@@ -25,14 +25,10 @@
   }
 
   .article-links {
-    display: flex;
-    flex-direction: column;
     &__two-col {
       @media (min-width: $bp-md) {
-        display: grid;
```
Comment (danielfmiranda) [inline source/sass/components/article-page.scss:32]: removing old css grid that I added last PR, and using column count instead.

**Review by sabrinang — APPROVED** [review]
It works well! The columns are balanced, the various link amounts arrange nicely and various line lengths work too. Thanks for the update Daniel!

**Review by mmmavis — CHANGES_REQUESTED** [review]
Lovely! Thanks for setting up test pages and leaving inline comments. 

Let's not use BEM naming convention until the engineering team as a whole agrees to adapt to it for Foundation site codebase. Could you please update `article-summary-link__two-col` to something else? Thanks!

File: source/sass/components/article-page.scss [hunk 4]
```diff
@@ -56,18 +50,22 @@
...
   flex-basis: calc(50% - var(--spacing));
   flex-grow: 0;
   flex-shrink: 0;
-
   @include hover-focus-active {
     color: inherit;
   }
 }
 
+.article-summary-link-two-column {
```
Comment (danielfmiranda) [inline source/sass/components/article-page.scss:65]: if there are more than 5 items, the article summary links are then given the "Article-summary-link-two-column" class which makes them inline block, to appear with uniform spacing between them

File: source/sass/components/article-page.scss [hunk 5]
```diff
@@ -23,18 +23,12 @@
...
-    flex-direction: column;
-    &__two-col {
-      @media (min-width: $bp-md) {
-        display: grid;
-        grid-template-rows: auto auto auto auto;
-        grid-auto-flow: column;
-        grid-gap: 10px;
-      }
-    }
+.article-links-two-column {
```
Comment (danielfmiranda) [inline source/sass/components/article-page.scss:28]: if there are more than 5 items, the article links  div is then given the "Article-links-two-column" class which gives it a column count of two, only above the medium breakpoint. 

If it is tablet-sized or under, the links display block one above another.

**Comment by danielfmiranda** [comment]
Hi @mmmavis Thank you for the catch and for taking a look! I have updated the CSS names like requested :)

**Review by mmmavis — APPROVED** [review]
Woohoooo summary link section is looking sharp ✨

**Pull request merged**